python/cudf/cudf/_lib/interop.pyx

```diff
@@ -66,7 +66,7 @@ def to_dlpack(list source_columns):
     return pycapsule.PyCapsule_New(
         dlpack_tensor,
         'dltensor',
-        dlmanaged_tensor_pycapsule_deleter
+        <pycapsule.PyCapsule_Destructor>dlmanaged_tensor_pycapsule_deleter
```

Comment:
Note that we're following the advice in [this StackOverflow post](https://stackoverflow.com/questions/59708440/cython-declare-a-pycapsule-destructor-in-pyx-file) about how to define the deleter, but apparently Cython 3.0.0 is less lax (or has a bug?) about how it accepts this and does require an explicit cast.

Reply:
Rather than an explicit cast, the fix is to define the `dlmanaged_tensor_pycapsule_deleter` function below with `noexcept`.

Function types must match up to their exception value. In Cython 3.0, non-extern functions (here, `dlmanaged_tensor_pycapsule_deleter) are implicitly declared with an exception value, and extern functions (here, `pycapsule.PyCapsule_Destructor`) are implicitly declared `noexcept`.

https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html?highlight=noexcept#error-return-values

Reply:
@shwina Amazing, thank you. I’ll try that fix!

Reply:
@shwina This seemed to work. I'll resolve this conversation and update the PR title/description.